### DISCUSSION TITLE
Using Membrane to get a frame from a video file?

### DISCUSSION THREAD

**Discussion by kipcole9** [membraneframework/membrane-core#799] [Q&A]
In my [Image](https://github.com/elixir-image/image) I'd like to be able to grab a video frame without having to jump out to eVision (OpenCV) or using FFMPEG natively. It *feels* like Membrane has all the pieces but I'm not sure where to start.

Let's assume for now that it's a video stream in a `.mp4` container.  How (can I?) open the container, access the stream, seek to a frame (or milliseconds) and get the video frame (I don't care what format the frame data is in, as long as I can detect what the format is).

Possible?

**Comment by mat-hek** [comment]
Hi @kipcole9, Membrane doesn't have first-class support for seeking, so you'd need to skip the stream until your conditions are met. Another option would be to use https://hexdocs.pm/membrane_mp4_plugin/Membrane.MP4.Container.html#parse!/1 directly. It returns the parsed MP4 structure, which you can use to manually find the size and offset of the frame you need. Let me know if you need help with either approach ;)
